File: atcoder/BeginnersSelection/06.ABC088B - Card Game for Two/golang/main.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var max int
	fmt.Scan(&max)

	cards := scanNums(max)
	sort.Sort(sort.Reverse(sort.IntSlice(cards)))

	var alice, bob int
    for i := 0; i < len(cards); i++ {
		if i%2 == 0 {
			alice +=  cards[i]
		} else {
			bob +=  cards[i]
		}
	}
	
	fmt.Println(alice - bob)
}

func scanNums(len int) (nums []int) {
    var num int
    for i := 0; i < len; i++ {
        fmt.Scan(&num)
        nums = append(nums, num)
    }
    return
}

File: atcoder/BeginnersSelection/06.ABC088B - Card Game for Two/rust/main.rs
fn main() {
    let _max = read::<u32>();
    let mut cards = read_vec::<u32>();

    cards.sort();
    cards.reverse();

    let mut alice = 0;
    let mut bob = 0;
    for (i, card) in cards.iter().enumerate() {
        match i%2 {
            0 => alice += *card,
            _ => bob += *card,
        }
    }

    println!("{}", alice - bob)
}

fn read<T: std::str::FromStr>() -> T {
    let mut s = String::new();
    std::io::stdin().read_line(&mut s).ok();
    s.trim().parse().ok().unwrap()
}

fn read_vec<T: std::str::FromStr>() -> Vec<T> {
    read::<String>().split_whitespace()
        .map(|e| e.parse().ok().unwrap()).collect()
}